### PR TITLE
Remove `@babel/plugin-transform-regenerator`

### PR DESCRIPTION
**Pull request by tido64**
## Summary

Hermes added support for generators in 0.2.0, and JSC is ES6 feature complete since [r202125](https://webkit.org/blog/6756/es6-feature-complete/).

## Test plan

n/a

### CODE REVIEW

**Comment by tido64** [comment]
@motiz88: It'd be great if you could make sure this doesn't break Meta internally.

**Comment by motiz88** [comment]
Awesome - can you please run Prettier, remove `@babel/plugin-transform-regenerator` from `package.json` and update `yarn.lock` accordingly?

**Comment by motiz88** [comment]
Looking good so far - I'm fixing up some minor Meta-internal things and planning to land this on Monday. Thanks for this, @tido64!

**Comment by newyankeecodeshop** [comment]
This is great. I believe you can also remove `babel-plugin-transform-for-of` since iteration is also supported in both runtimes. Hopefully the minifier supports the syntax too.

**Comment by rh389** [comment]
`uglify-es` and `terser` both fully support ES6 so I wouldn’t expect any issue for generators or for/of. `uglify-es` starts to get ugly for ES7+, which is part of the reason I’m looking to shift the default to `terser`.

**Comment by tido64** [comment]
image

@motiz88: Something's failing internally. Is it because of this change?

> This is great. I believe you can also remove `babel-plugin-transform-for-of` since iteration is also supported in both runtimes.

That's a great idea, but I think it should be a separate PR.

> > Hopefully the minifier supports the syntax too.
> 
> That is a good point - I'm not actually sure we have a test that would fail if it didn't. @tido64 would you mind checking to see what happens if you run this PR with `minify=true`? (cc @rh389 re: minifier stuff)

How do I turn this on in tests? I tried pasting some of the snapshots in [Terser's REPL](https://try.terser.org/) and [Uglify's demo](https://lisperator.net/uglifyjs/). Terser is fine with the generator function, but UglifyJS doesn't support it. I know uglify-es is supposed to support ES6, but it's deprecated and I see that people are trying to migrate to the former: https://github.com/facebook/metro/pull/661

Is this going to be a problem?

**Comment by rh389** [comment]
> I know uglify-es is supposed to support ES6, but it's deprecated and I see that people are trying to migrate to the former: https://github.com/facebook/metro/pull/661

There's a bit of history here - `uglify-js` originally didn't support ES6. The author created `uglify-es` (originally a `harmony` branch of `mishoo/UglifyJS2`) to address that, but that was itself [deprecated](https://www.npmjs.com/package/uglify-es) some time ago in favour of `uglify-js`, as `uglify-js` v3 *does* now support ES6 out of the box. I suspect that demo is just out of date. They all have CLIs:

```sh
➜ echo "function* myGenerator(myVar) { yield myVar + 1; }" | npx uglify-js -m
function*myGenerator(e){yield e+1}

➜  echo "function* myGenerator(myVar) { yield myVar + 1; }" | npx terser -m
function*myGenerator(e){yield e+1}

➜  echo "function* myGenerator(myVar) { yield myVar + 1; }" | npx uglify-es -m
function*myGenerator(e){yield e+1}
```

Terser was originally a fork of `mishoo/UglifyJS2` so it has a shared ancestry with `uglify-es`, and internally we've found it to be closer to `uglify-es` in output than `uglify-js@3` is (#661 turned out to be problematic internally), so we're working to make `terser` the metro default and we'll probably deprecate `metro-minifier-uglify`.

**TL;DR**: Nothing minifier-related to worry about for this PR.

**Comment by rh389** [comment]
> Something's failing internally. Is it because of this change?

Unrelated - React Native iOS tests are broken at the moment.
